CONTRIBUTING.md

```diff
@@ -1,51 +1,150 @@
+steps:
+
+1. Make sure to search for other users who may have had similar issues in the
+   past.  Search the repository's issues page (don't forget to search closed
+   issues), bring it up on Discord, use sites like StackOverflow, and exercise
+   your best Google practices.  If you can't find anything helpful, then go to
+   the next step.
+
+1. Make sure you can reproduce the issue, i.e. that you have some code that
+   illustrates the bug **every time** (or at least most of the time) it is
+   executed.
+   
+2. Make sure to clarify what behavior you expected, and how the actual behavior
+   differs from your expectation.
+
+3. Make sure to gather information about your environment, such as your
+   operating system, manim version, python version, and any stack traces that
+   the code may have generated (if applicable).
+   
+4. Only after you have gathered this information, please open an issue.  A
+   community member will (hopefully) respond and start a conversation to
+   address the issue.
+   
+5. Please give the community a reasonable amount of time before asking again,
+   or insisting on your issue.  Keep in mind that everyone is a volunteer.  If
+   you wait for a reasonable amount of time and you receive no response, feel
+   free to insist.
+
+
+## Contributing code
+
+Many ways of contributing will involve writing, reading, testing, or
+refactoring code.  As our repository is a Fork of [Manim by
+3b1b](https://github.com/3b1b/manim), contributing in this way can be a bit
+confusing.  Here is a short guide on how to do it.
+
+1. First, make a fork of this repository.  This creates your own copy of the
+   whole codebase on GitHub.
+
+2. Clone the repository you forked using the command below (bash/zsh).
 
-## General instructions
-
-As this is a Fork of [Manim by 3b1b](https://github.com/3b1b/manim), contributing can be a bit confusing. Because of that, here is a small guide on it. 
-
-1. First, make a fork of this repository.
-2. Then clone the repository you forked using the command below (bash/zsh).
+   ```sh
+     git clone <your-fork-url>
+     cd manim
+   ```
 
- ```sh
-   git clone <your-fork-url>
-   cd manim
- ```
+	Note: do not `git clone` the original ManimCommunity repository.  You must
+    clone your own fork.  After this step, there are three different
+    repositories to keep track of: the original ManimCommunity repo, your own
+    fork of it, and your local repository.
 
-3. Now that you have cloned the repo, before making any changes, you have to enter the commands below in order to be able to contribute to Manim Community.
+3. You have to make `git` be aware of the three repositories, by entering the
+   commands below.
 
    ```sh
-   git remote add fork https://github.com/ManimCommunity/manim.git
-   git fetch fork
-   git checkout -b <your-branch-name> fork/master
+   git remote add upstream https://github.com/ManimCommunity/manim.git
+   git pull upstream master
```

Comment:
```suggestion
   git remote add community https://github.com/ManimCommunity/manim.git
   git pull community master
```

Nitpick: I think naming the remote `community` over `upstream` will provide more clarity as to what exactly the remote is.

Reply:
I disagree here. I think upstream is more standard (and it's also shorter).